src/main/java/org/cirdles/topsoil/data/NumberField.java

```diff
@@ -43,7 +44,13 @@ public String toString(Number number) {
 
             @Override
             public Number fromString(String string) {
-                return Double.valueOf(string);
+                
+               //Check if the string parameter is a number
+                if (string.matches("\\d+(\\.\\d+)?")) {
```

Comment:
What about `10e-10`? There's actually a regex that `Double.valueOf` (or maybe `Double.parse`) uses that's listed in the Javadocs.